src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala

```diff
@@ -209,14 +212,21 @@ object RouteGenerator {
     case HttpMethods.PATCH => patch
   }
 
-  private def corsSettings(allowedMethods: immutable.Seq[HttpMethod]): CorsSettings =
-    CorsSettings.defaultSettings.withAllowedMethods(allowedMethods)
-
-   def requestLocales(headers: Map[String, String]): List[Locale] =
+  def requestLocales(headers: Map[String, String]): List[Locale] =
      headers.get("accept-language") match {
       case Some(localeString) if localeString.nonEmpty => LanguageRange.parse(localeString).asScala
         .map(language => Locale.forLanguageTag(language.getRange)).toList
       case _ => Nil
     }
 
+  private def corsRejectionHandler(request: AkkaHttpRequest, accessLogIdGetter:Option[AkkaHttpRequest => String]): RejectionHandler =
+    RejectionHandler
+      .newBuilder()
+      .handleAll[javadsl.CorsRejection] { rejections =>
+        val causes = rejections.map(_.cause.description).mkString(", ")
+        accessLogIdGetter.foreach(g => AccessLog.logAccess(request, g(request), StatusCodes.Forbidden))

```

Comment:
We still want to write to the access log even if accessLogIdGetter is None.

Reply:
At this point in the code if accessLogIdGetter is None then access logging has not been enabled for this service.

Reply:
See line https://github.com/oracle/wookiee-akka-http/blob/2.0.x/src/main/scala/com/webtrends/harness/component/akkahttp/routes/AkkaHttpEndpointRegistration.scala#L56